src/Adapter/MSTest.CoreAdapter/Resources/Resource.resx

```diff
@@ -368,4 +368,19 @@ Error: {1}</value>
   <data name="OlderTFMVersionFoundClassCleanup" xml:space="preserve">
     <value>An older version of MSTestV2 package is loaded in assembly, test cleanup methods might not run as expected. Please make sure all your test projects references MSTest pacakges newer then version 2.2.8.</value>
   </data>
+  <data name="CannotExpandIDataSourceAttribute" xml:space="preserve">
+    <value>Exception occurred while expanding IDataSource rows from attribute on "{0}.{1}": {2}</value>
```

Comment:
```suggestion
    <value>Exception occurred while expanding "IDataSource" rows from attribute on "{0}.{1}". Error: {2}</value>
```

Reply:
But maybe I would remove the `Error: {2}` and print 2 warnings or add the relevant test info (i.e. `{0}.{1}`) inside the other 2 messages.

Reply:
It's not an error, it's the continuation of the message. There will be more edge cases here, so it's easier on localization to do this way. A standard part, and a detail part.

Reply:
Yeah error was probably not the best wording. Let's continue as-is but IMO the full message reads weird.